### PR TITLE
[Feature] Optimized Point Cloud Rendering

### PR DESCRIPTION
**Pull request by TomaszTB**
This PR introduces two efficient point cloud renderers. One that takes in RawImages as input, and the other takes in list/array of points. 

The existing point cloud renderer is renamed and deprecated. I plan on removing usages of the old renderer in future PRs.

### CODE REVIEW

File: ihmc-graphics/src/libgdx/java/us/ihmc/rdx/shader/RDXShader.java [hunk 1]
```diff
@@ -22,12 +22,21 @@ public RDXShader(Class<?> clazz)
    }
 
    public void create()
+   {
+      create(null, null);
+   }
+
+   public void create(String[] vertexFlags, String[] fragmentFlags)
```
Comment (TomaszTB) [inline ihmc-graphics/src/libgdx/java/us/ihmc/rdx/shader/RDXShader.java:29]: Added support for adding #define flags to the shaders during runtime

File: ihmc-high-level-behaviors/src/libgdx/resources/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.glsl
Comment (TomaszTB) [inline ihmc-high-level-behaviors/src/libgdx/resources/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.glsl:1]: This guy performs what the [PinholePinholeColoredPointCloud.cl](https://github.com/ihmcrobotics/ihmc-open-robotics-software/blob/1df18d96b4368d25cc46e3626c1c1f9e72e5a151/ihmc-high-level-behaviors/src/libgdx/resources/openCL/PinholePinholeColoredPointCloud.cl) files does and the rendering part. This significantly reduces data transfer between CPU and GPU.

File: ihmc-graphics/src/libgdx/resources/us/ihmc/rdx/RDXPointCloudRenderer.glsl
Comment (TomaszTB) [inline ihmc-graphics/src/libgdx/resources/us/ihmc/rdx/RDXPointCloudRenderer.glsl:1]: Removed support for passing in a color and size for each individual point, as that requires more data transfer. Instead, you can specify a default point color for all points, or use a gradient coloring mode. That's how we use the renderer most often anyway.

File: ihmc-graphics/src/libgdx/java/us/ihmc/rdx/AbstractRDXPointCloudRenderer.java
Comment (TomaszTB) [inline ihmc-graphics/src/libgdx/java/us/ihmc/rdx/AbstractRDXPointCloudRenderer.java:1]: This class contains common code of the point cloud renderers, and some abstract methods which must be defined to initialize the shader correctly.

File: ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      // Ensure correct length and initialize vertices buffer
+      if (renderable.meshPart.size != pixelCount)
+      {
+         renderable.meshPart.size = pixelCount;
+         verticesBuffer.limit(pixelCount);
+      }
+
+      // Wrap the vertices buffer into a pointer, than into a Mat
+      FloatPointer verticesPointer = new FloatPointer(verticesBuffer);
+      Mat verticesMat = new Mat(height, width, opencv_core.CV_32FC1, verticesPointer);
```
Comment (ds58) [inline ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.java:89]: maybe add a comment about why we do this

**Review by ds58 — APPROVED** [review]

File: ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.java (hunk 2 above)
Comment (TomaszTB) [reply ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/graphics/RDXRawImagePointCloudRenderer.java:89]: Added a comment. Let me know if I should edit it.

File: ihmc-graphics/src/libgdx/java/us/ihmc/rdx/shader/RDXShader.java (hunk 1 above)
Comment (calvertdw) [reply ihmc-graphics/src/libgdx/java/us/ihmc/rdx/shader/RDXShader.java:29]: whoa

**Review by calvertdw — APPROVED** [review]

**Pull request merged**